app/src/main/java/de/westnordost/streetcomplete/tangram/QuestsMapFragment.java

```diff
@@ -176,12 +178,23 @@ private void zoomAndMoveToContain(ElementGeometry g)
 
 	private LngLat getCenterWithOffset(ElementGeometry geometry)
 	{
-		LngLat top = controller.screenPositionToLngLat(new PointF(0, questTopOffset));
-		LngLat bottom = controller.screenPositionToLngLat(new PointF(0, questBottomOffset));
-		LngLat offset = new LngLat(0,0);
-		offset.latitude = (top.latitude - bottom.latitude) / 2;
-		offset.longitude = (top.longitude - bottom.longitude) / 2;
 		LngLat pos = TangramConst.toLngLat(geometry.center);
+		LngLat offset = new LngLat(0,0);
+
+		if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT)
+		{
+			LngLat top = controller.screenPositionToLngLat(new PointF(0, questOffset.top));
+			LngLat bottom = controller.screenPositionToLngLat(new PointF(0, questOffset.bottom));
+			offset.latitude = (top.latitude - bottom.latitude) / 2;
+			offset.longitude = (top.longitude - bottom.longitude) / 2;
+		}
+		else if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE)
+		{
+			LngLat left = controller.screenPositionToLngLat(new PointF(questOffset.left, 0));
+			LngLat bottom = controller.screenPositionToLngLat(new PointF(0, questOffset.bottom));
+			offset.latitude = (left.latitude - bottom.latitude) / 2;
+			offset.longitude = (left.longitude - bottom.longitude) / 2;
+		}
```

Comment:
This logic does not belong here. The `questOffset` `Rect` should at this point already exactly have the offset that should be used and that offset should simply be applied.

Reply:
What do you mean? I'm sorry, but I don't get your point.